TemporaryProjects/StartPageExtender.cs

```diff
@@ -0,0 +1,157 @@
+                    guid == VSConstants.StandardToolWindows.StartPage &&
+                    ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocView, out var windowPane)) &&
+                    windowPane?.GetType().GetProperty(nameof(WindowPane.Content)).GetValue(windowPane) is IVsUIWpfElement wpfElement &&
+                    ErrorHandler.Succeeded(wpfElement.GetFrameworkElement(out var element)) &&
+                    element is FrameworkElement frameworkElement &&
+                    frameworkElement.FindName("ContentHost") is Decorator contentHost)
+                {
+                    SetupStartPage(contentHost);
+                }
+            }
+        }
+
+        private void SetupStartPage(Decorator contentHost)
+        {
+            contentHost.LayoutUpdated += LayoutUpdated;
+
+            void LayoutUpdated(object sender, EventArgs e)
+            {
+                var recentProjectsPanel = FindRecentProjectsPanel(contentHost);
+                if (recentProjectsPanel != null)
+                {
+                    AddNewTemporaryProjectButton(recentProjectsPanel);
+                    contentHost.LayoutUpdated -= LayoutUpdated;
+                }
+            }
+        }
+
+        private static Grid FindRecentProjectsPanel(Decorator contentHost)
+        {
+            if (contentHost.Child is ContentControl start &&
+                start.Template.FindName("NewProjectsListView", start) is ContentControl newProjectsListView &&
+                newProjectsListView.Template.FindName("RecentProjectsPanel", newProjectsListView) is Grid recentProjectsPanel)
+            {
+                return recentProjectsPanel;
+            }
+
+            return null;
+        }
+
+        private void AddNewTemporaryProjectButton(Grid recentProjectsPanel)
+        {
+            var button = new NewTemporaryProjectButton(dte)
+            {
+                Style = ((Button)recentProjectsPanel.FindName("MoreTemplatesButton")).Style
+            };
+
+            recentProjectsPanel.RowDefinitions.Add(new RowDefinition() { Height = GridLength.Auto });
+            Grid.SetRow(button, recentProjectsPanel.RowDefinitions.Count - 1);
+
+            recentProjectsPanel.Children.Add(button);
+        }
+
+        private sealed class NewTemporaryProjectButton : Button
+        {
+            private readonly DTE dte;
+
+            public NewTemporaryProjectButton(DTE dte)
+            {
+                this.dte = dte;
+            }
+
+            public override void OnApplyTemplate()
+            {
+                base.OnApplyTemplate();
+
+                var textBlock = (TextBlock)GetTemplateChild("MoreTemplatesText");
+                textBlock.Text = "Create new temporary project...";
+            }
+
+            protected override void OnClick()
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+
+                base.OnClick();
+
+                NewTempProjectCommand.Execute(dte);
```

Comment:
I think we can use something like the following to raise the command using the Visual Studio command infrastructure:
```cs
object _ = null;
dte.Commands.Raise(NewTempProjectCommand.CommandSet.ToString("B"), NewTempProjectCommand.CommandId, _, _);
```

That way if someone monitors executing commands (using something like `LearnTheShortcut`), our command will appear.

Reply:
Thanks, this is exactly what I was looking for. I tried using `ExecuteCommand`, which didn't work, and couldn't find the right API to do this.

Reply:
Fixed in a604808f335864c9b65f91af588e6cf5d1332daa